### PR TITLE
docs: ctags-client-tools(7): write about json output format

### LINKED ISSUE #2964: JSON output format
Command:
```
$ ctags --fields=* --output-format=json -f ctags.json file.c
```

The output isn't valid json. It needs to be a JSON array.
So, the output needs enclosing "[ ]" and commas to separate the objects.

### PR DESCRIPTION
**Pull request by masatake**
Close #2964.

### CODE REVIEW

File: man/ctags-client-tools.7.rst.in [hunk 1]
```diff
@@ -462,6 +462,84 @@ client tool could distinguish them by the ``TAG_OUTPUT_EXCMD`` pseudo tag, it's
...
+"``--output-format=json --list-fields``" options list the fields.
+
+A key starting with ``_`` represents meta information of the JSON
+line.  Currently only ``_type`` key is used. If the value for the key
+is ``tag``, the JSON line represents a normal tag. If the value is
+``ptag``, the line represents a pseudo-tag.
+
+The output format can be changed in the
+future. ``JSON_OUTPUT_VERSION`` pseudo-tag provides a change
+client-tools to handle the changes.  Current version is "0.0". A
```
Comment (hirooih) [inline man/ctags-client-tools.7.rst.in:496]: This is the time to change version "0.0" to "1.0" or "0.1".

File: man/ctags-client-tools.7.rst.in [hunk 2]
```diff
@@ -462,6 +462,84 @@ client tool could distinguish them by the ``TAG_OUTPUT_EXCMD`` pseudo tag, it's
...
+client-tool can extract the version with ``path`` key from the
+pseudo-tag.
+
+The JSON output format is newly designed and has no limitation found
+in the default tags file format.
+
+* The values for ``kind`` key are represented in long-name flags.
+  No one-letter is here.
+
+* scope kinds and scope names have distinguished keys: ``scope`` and ``scopeKind``.
```
Comment (hirooih) [inline man/ctags-client-tools.7.rst.in:506]: "Scope names and scope kinds ..."  (capitalize and change the order as ``scope`` and ``scopeKind``).

File: man/ctags-client-tools.7.rst.in [hunk 3]
```diff
@@ -462,6 +462,84 @@ client tool could distinguish them by the ``TAG_OUTPUT_EXCMD`` pseudo tag, it's
...
+  They are combined in the default tags file format.
+
+Data type used in a field
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Values for the most of all keys are represented in JSON string type.
+However, some of them are represented in integer type and/or boolean type.
```
Comment (hirooih) [inline man/ctags-client-tools.7.rst.in:512]: "... represented in string, integer, and/or boolean type."

**Review by hirooih — COMMENTED** [review]
I put some minor comments.

When you move contents from a file to another, please commit them before changing the contents.  It help reviewers to know what are changes.

File: man/ctags-client-tools.7.rst.in (hunk 1 above)
Comment (masatake) [reply man/ctags-client-tools.7.rst.in:496]: The output format is not changed. So I will not change the number.

File: man/ctags-client-tools.7.rst.in (hunk 2 above)
Comment (masatake) [reply man/ctags-client-tools.7.rst.in:506]: I see. Thank you.

File: man/ctags-client-tools.7.rst.in (hunk 3 above)
Comment (masatake) [reply man/ctags-client-tools.7.rst.in:512]: I see. Thank you.

File: man/ctags-client-tools.7.rst.in (hunk 1 above)
Comment (hirooih) [reply man/ctags-client-tools.7.rst.in:496]: I see. I misunderstood.

**Review by hirooih — APPROVED** [review]
LGTM

**Comment by masatake** [comment]
@hirooih, thank you for reviewing.

**Pull request merged**